### PR TITLE
chore: fix failing tests 

### PR DESCRIPTION
**Pull request by FuzzB0t**
This PR adds some fixes to the unit and e2e tests + adds fixes to snx ci

Fixes https://linear.app/usecannon/issue/CAN-176/make-ci-green

### CODE REVIEW

File: packages/cli/test/e2e/scripts/non-interactive/downstream-ci-synthetix-v3.sh [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 git clone --depth=1 https://github.com/Synthetixio/synthetix-v3
 cd synthetix-v3
+yarn link $CANNON_REPO_DIR/packages/cli
+yarn link $CANNON_REPO_DIR/packages/hardhat-cannon
```
[the remark this answers is not in the document]
Comment (FuzzB0t) [reply packages/cli/test/e2e/scripts/non-interactive/downstream-ci-synthetix-v3.sh:6]: When the builder `yarn link` fails due to circular dependencies between packages.

[the remark this answers is not in the document]
Comment (FuzzB0t) [reply packages/cli/test/e2e/scripts/non-interactive/downstream-ci-synthetix-v3.sh:6]: I think its necessary technically as part of the test to cover all of the packages but still need to work through how to set that up.

[the remark this answers is not in the document]
Comment (dbeal-eth) [reply packages/cli/test/e2e/scripts/non-interactive/downstream-ci-synthetix-v3.sh:6]: I think running `yarn link` before running `yarn` may erase the previously set `link`s... can you confirm this is not the case?

**Review by dbeal-eth — CHANGES_REQUESTED** [review]

**Comment by dbeal-eth** [comment]
also, some of hte tests still appear to be failing on ci. is there anything we can do to improve this?

File: packages/cli/test/e2e/scripts/non-interactive/downstream-ci-synthetix-v3.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (FuzzB0t) [reply packages/cli/test/e2e/scripts/non-interactive/downstream-ci-synthetix-v3.sh:6]: I dont think `yarn link` will break previous links. Each linked package is managed independently, so linking one package shouldn't interfere with links to other packages.

**Review by dbeal-eth — APPROVED** [review]

**Pull request merged**